docs/streams/developer-guide.html

```diff
@@ -209,6 +210,33 @@
 
     <h3><a id="streams_dsl" href="#streams_dsl">High-Level Streams DSL</a></h3>
 
+    The Kafka Streams DSL is the recommended API for most users – and notably for starters – because most data processing use cases
+    can be expressed in just a few lines of DSL code. Also, compared to the Kafka Streams Processor API, only the DSL supports:
+
+    <ul>
+      <li>Built-in abstractions for streams and tables in the form of <code>KStream</code>, <code>KTable</code>, and <code>GlobalKTable</code>.
+	Having first-class support for streams
+	and tables is crucial because, in practice, most use cases require not just either streams or databases/tables, but a combination of both.
+	For example, if your use case is to create a customer 360-degree view that is updated in real-time, what your application will be
+	doing is transforming many input streams of customer-related events into an output table that contains a continuously updated
+	360-degree view of your customers.</li>
+
+      <li>Declarative, functional programming style with stateless transformations (e.g. <code>map</code> and <code>filter</code>)
+	as well as stateful transformations such as aggregations (e.g. <code>count</code> and <code>reduce</code>),
+	joins (e.g. <code>leftJoin</code>), and windowing (e.g. session windows). </li>
+
+    <li>And more, as described in the following sections.</li>
+    </ul>
+    With the DSL, users can define processor topologies, i.e. the logical processing plan, in their application by specifying
+    one or more input streams that are being read from Kafka topics, followed by composing one or more transformations on these
+    streams, and finally writing the resulting output streams back to Kafka topics or exposing the processing results of their
+    application directly to other applications through interactive queries (e.g., via a REST API). Once the application is run,
+    the defined processor topologies are being continuously executed, that is, the processing plan is put into action. 
+      
+    
+    In the subsequent sections we provide a step-by-step guide for writing a stream processing application using the DSL.
+
+    
```

Comment:
The duality will move to the core concepts as part of a separate PR. I'm not touching in this PR.